_source/_posts/2022-08-09-spring-boot-saml.adoc

```diff
@@ -0,0 +1,574 @@
+---
+layout: blog_post
+title: Get Started with Spring Boot and SAML
+author: matt-raible
+by: advocate
+communities: [java]
+description: "Learn how to build a Spring Boot application that authenticates against Okta and Auth0 with Spring Security's SAML support."
+tweets:
+  - "Need SAML integration in your Spring Boot application? Get Started with Spring Boot and SAML using Okta!"
+  - "Spring Boot + SAML + Okta = 💙! Learn more →"
+tags: [spring-boot, saml, spring-security]
+type: conversion
+github: https://github.com/oktadev/okta-spring-boot-saml-example
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+Spring is a long-time friend to enterprise companies throughout the world. When Spring Boot came along in 2014, it greatly simplified configuring a Spring application. This led to widespread adoption and continued investment in related Spring projects.
+
+One of my favorite Spring projects is Spring Security. In most cases, it simplifies web security to just a few lines of code. HTTP Basic, JDBC, JWT, OpenID Connect/OAuth 2.0, you name it&mdash;Spring Security does it!
+
+You might notice I didn't mention SAML as an authentication type. That's because I don't recommend it. The specification for SAML 2.0 was published in March 2005, before smartphones or smart devices even existed. OpenID Connect (OIDC) is much easier for developers to use and understand. Using SAML in 2022 is like implementing a web service using WS-* instead of REST.
+
+My recommendation: just use OIDC.
+
+If you _must_ use SAML with Spring Boot, this tutorial should make it quick and easy.
+
+**Prerequisites:**
+
+- https://sdkman.io/[SDKMAN] (for Java 17)
+
+toc::[]
+
+== What is SAML?
+
+**S**ecurity **A**ssertion **M**arkup **L**anguage is an XML-based way of doing web authentication and authorization. It works cross-domain, so SaaS applications and other enterprise software often support it.
+
+Nick Gamb has an excellent overview in https://www.youtube.com/watch?v=l-6QSEqDJPo[A Developer's Guide to SAML].
+
+++++
+{% youtube l-6QSEqDJPo %}
+++++
+
+See Spring Security's documentation for a https://docs.spring.io/spring-security/reference/servlet/saml2/login/overview.html[SAML 2.0 Login Overview].
+
+== Add a SAML application on Okta
+
+To begin, you'll need an Okta developer account. You can create one at https://developer.okta.com/signup[developer.okta.com/signup] or install the https://cli.okta.com[Okta CLI] and run `okta register`.
+
+Then, log in to your account and go to *Applications* > *Create App Integration*. Select *SAML 2.0* and click *Next*. Name your app something like `Spring Boot SAML` and click *Next*.
+
+Use the following settings:
+
+* Single sign on URL: `\http://localhost:8080/login/saml2/sso/okta`
```

Comment:
```suggestion
* Single sign-on URL: `\http://localhost:8080/login/saml2/sso/okta`
```
nit

Reply:
Unfortunately, this is how it's spelled in the Okta Admin Console. 😞

<img width="707" alt="Screen Shot 2022-08-04 at 13 17 52" src="https://user-images.githubusercontent.com/17892/182934966-2c4dd5fc-4379-443b-803f-aaf3c2ebc5f9.png">